### PR TITLE
Ceres: prepare AENS auction changes

### PR DESCRIPTION
**Pull request by hanssv**
This introduce the necessary refactorings, and a logic mechanism for shorter auction extensions.  Related to #3853

This PR is supported by the Æternity Crypto Foundation

### CODE REVIEW

**Comment by marc0olo** [comment]
@hanssv what kind of params do we need to discuss now again? can you summarize that shortly? would be great :)

**Review by uwiger — APPROVED** [review]
otherwise, LGTM

**Comment by hanssv** [comment]
> Shouldn't the `ct:pal/2` calls be replaced by calls to `ct:log/2`?

That test suite seems to be all `ct:pal`?!

**Review by uwiger — DISMISSED** [review]

**Review by uwiger — APPROVED** [review]

**Review by seanhinde — APPROVED** [review]
